installation_policies/xSDK_spack_variant_guidelines.md

```diff
@@ -0,0 +1,88 @@
+
+## List of recommended variants
+
+The following options must be expressed through Spack variants, where relevant.
+
+
+**1. Index size**
+
+This option concerns index sizes (or compile-time determined integer sizes).
+
+We recommend using the boolean `int64` variant name to set index sizes to 64-bits. It is expected that `~int64` will result in 32-bit integer size.
+
+`int64`: Index sizes (or compile-time determined integer sizes) are set to 64 bits.
+
+When different options are required, an alternative variant (either boolean or multi-valued) may be used. Please refer to the [Spack variant review](https://github.com/xsdk-project/xsdk-community-policies/wiki/Spack-variant-review) for guidance.
+
+---
+
+**2. Precision**
+
+This concerns the size of compile-time configured floating-point variables.
+
+We recommend using multi-valued variant `precision`.
+
+`precision`: The multi-valued variant `precision` can accept multiple precision descriptions. The names `single` and `double` are recommended for 32-bit/64-bit respectively. Other options may include `half` for 16-bit, `extended` for 80-bit or `quad` for 128-bit.
+
+---
+
+**3. Shared/Static libraries**
+
+These options determine how libraries are built.
+
+**_Note_**
+There has been discussion about adopting a multi-value variant for this purpose, which may better cover the possible permutations. However, the following recommendations have been determined based on a review of current usage. It is possible this will change in the future.
+
+We recommend using one or more of following boolean variants which have been adopted widely in Spack:
+
+- `shared`: A boolean variant. If True, then shared libraries will be built.
+- `static`: A boolean variant. If True, then static libraries will be built.
+- `pic`: A boolean variant. If True, then produce position independent code.
+
+The meaning and combination of these values is expected to follow conventions given below (based on CMake):
+
+If `shared` is the only option present, then enabling this option is expected to build shared libraries *instead* of static, and these will be built using _pic_ (position independent code). If False, then static libraries will be built without pic.
+
+If the package author wishes to provide an option to build both static and shared libraries, then use both `static` and `shared` variants, and set both to True.
+
+To build shared libraries but without _pic_, then use `shared` and `pic`, and set `pic` to False.
+
+Permutations of package variants that are unsupported or infeasible should be denoted in a `conflict` statement. E.g. `conflicts('~static~shared')`
+
+---
+
+**4. Debug/build_type**
+
+This option configures debugging symbols and optimization.
+
+We recommend using the multi-valued variant `build_type`.
```

Comment:
This is listed as M16 - as mandatory policy. Here is says `recommended`. Should M16 be moved to R-something?

Reply:
The exact M16 language is

> In Spack, the debug build type should be enabled via the variants recommended in the [xSDK Spack variant guidelines](../installation_policies/xSDK_spack_variant_guidelines.md).

We could change "should be" to "should be, if applicable" and then the policy can remain mandatory.